ballerina/metadata_types.bal

```diff
@@ -22,21 +22,80 @@ import ballerina/sql;
+    map<JoinMetadata> joinMetadata?;
 |};
 
-# Represents the metadata of a simple field in the entity.
+
+# Represents the metadata associated with a simple field in the entity record.
 #
-# + 'type - The data type of the field  
 # + columnName - The name of the SQL table column to which the field is mapped
-# + autoGenerated - If true, the value of the field is not defined on creation 
-#                   but is instead generated by the SQL database
-public type FieldMetadata record {|
-    typedesc 'type;
+public type SimpleFieldMetadata record {|
     string columnName;
-    boolean autoGenerated = false;
 |};
+
+# Represents the metadata associated with a field from a related entity.
+#
+# + relation - The relational metadata associated with the field
+public type EntityFieldMetadata record {|
+    RelationMetadata relation;
+|};
+
+# Represents the metadata associated with a field of an entity.
+# Only used by the generated persist clients and `persist:SQLClient`.
+#
+public type FieldMetadata SimpleFieldMetadata|EntityFieldMetadata;
+
+# Represents the metadata associated with a relation.
+# Only used by the generated persist clients and `persist:SQLClient`.
+#
+# + entityName - The name of the entity represented in the relation  
+# + refField - The name of the referenced column in the SQL table
+
+public type RelationMetadata record {|
+    string entityName;
+    string refField;
+|};
+
+# Represents the metadata associated with performing an SQL `JOIN` operation.
+# Only used by the generated persist clients and `persist:SQLClient`.
+#
+# + entity - The name of the entity that is being joined  
+# + fieldName - The name of the field in the `entity` that is being joined  
+# + refTable - The name of the SQL table to be joined  
+# + refColumns - The names of the referenced columns of the referenced table
+# + joinColumns - The names of the join columns
+# + joinTable - The name of the joining table used for a many-to-many relation
+# + joiningRefColumns - The names of the referenced columns in the joining table     
+# + joiningJoinColumns - The names of the join columns in the joining table     
+# + 'type - The type of the relation
+public type JoinMetadata record {|
+    typedesc<record {}> entity;
+    string fieldName;
+    string refTable;
+    string[] refColumns;
+    string[] joinColumns;
+    string joinTable?;
+    string[] joiningRefColumns?;
+    string[] joiningJoinColumns?;
+    JoinType 'type;
+|};
+
+# Represents the type of the relation used in a `JOIN` operation.
+# Only used by the generated persist clients and `persist:SQLClient`.
+#
+# + ONE_TO_ONE - The association type is a one-to-one association
+# + ONE_TO_MANY - The entity is in the 'one' side of a one-to-many association
+# + MANY_TO_ONE - The entity is in the 'many' side of a one-to-many association
+public enum JoinType {
+    ONE_TO_ONE,
+    ONE_TO_MANY,
+    MANY_TO_ONE
+}
+
+//regex for "type" keyword following by string of characters
```

Comment:
```suggestion
```

Reply:
Fixed in [3ac4df4](https://github.com/ballerina-platform/module-ballerina-persist/pull/155/commits/3ac4df4447e4c445dda487eb932865f44b7f670c)